packages/atlas-service/src/main.ts

```diff
@@ -63,15 +59,15 @@ export class CompassAuthService {
   ): Promise<Response> => {
     await this.initPromise;
     this.throwIfNetworkTrafficDisabled();
-    throwIfAborted(init.signal as AbortSignal);
+    throwIfAborted(init.signal ?? undefined);
     log.info(
       mongoLogId(1_001_000_299),
       'AtlasService',
       'Making an unauthenticated fetch',
       { url }
     );
     try {
-      const res = await nodeFetch(url, {
+      const res = await net.fetch(url, {
```

Comment:
I don't know if this is a chance we want to make at this point. We'll be picking up https://jira.mongodb.org/browse/COMPASS-8011 shortly, and I think proxy support for Atlas login is going to be a requirement; for the OIDC plugin code, that would happen through Node.js `https.Agent` instances, which node-fetch supports, but neither of the built-in `fetch()` functions will.

I'm still thinking through what this will actually look like in the long run, but for now, I'd stick to node-fetch until we know what our story is going to be?

Reply:
I totally lack the knowledge what the requirements for this project will be, but I would imagine that enterprise customers would want system proxy already configured on their machines to work over something like providing a special configuration for Compass, which is something that works out of the box when using Electron / Chromium network stack (fetch in the renderer, net in the main). I'm pretty sure that this is why for all the issues that users reported to us wrt of Atlas login failing they were able to go through the sign in process in the browser, but oidc-plugin failed to fetch.

I don't want to push for a specific solution if you feel strong that we need to use Node.js http (through node-fetch or not) here, but seems like allowing oidc-plugin to use the network stack of the runtime we're developing for that already has built-in behavior for supporting http proxies might be an approach that would be more aligned with the runtime we're using. Otherwise we should probably abstract the network stack into a shared module so that the renderer behavior does not differentiate from the main process, it feels to me like this is a very non obvious inconsistency in the applicaiton behavior otherwise.

Reply:
Clarifying, are you suggesting to wait on providing a `https.Agent` with the `ca` for https://jira.mongodb.org/browse/COMPASS-8011 and keep the fetch as is for now? Then in this work we only do the plugin `ca` passing?

Reply:
> I would imagine that enterprise customers would want system proxy already configured on their machines to work over something like providing a special configuration for Compass

Maybe, yeah. I would still expect from customers that they'd want to have the option to configure this in our tools, though. But this isn't a decision that's already made at this point.

> which is something that works out of the box when using Electron / Chromium network stack (fetch in the renderer, net in the main).

As long as everything we're using supports this stack, yes – but that doesn't apply to our `oidc-plugin`, for example.

> but seems like allowing oidc-plugin to use the network stack of the runtime we're developing for that already has built-in behavior for supporting http proxies might be an approach that would be more aligned with the runtime we're using

That would probably require forking openid-client, or replacing it. Not impossible, and it's annoying that we can't configure the implementation in it for HTTP requests at all right now, only pass different options.

> Otherwise we should probably abstract the network stack into a shared module so that the renderer behavior does not differentiate from the main process, it feels to me like this is a very non obvious inconsistency in the applicaiton behavior otherwise.

Yeah, I can see the downsides of this approach but my current intuition is that this is a route we'd go down.

> Clarifying, are you suggesting to wait on providing a `https.Agent` with the `ca` for https://jira.mongodb.org/browse/COMPASS-8011 and keep the fetch as is for now? Then in this work we only do the plugin `ca` passing? Thanks for the context!

Err, so, I'm saying that it might be better in this PR to pass the `ca` like we do in `oidc-plugin`, i.e. using a custom `https.Agent`, and not switch out the `fetch()` implementation as well (as this PR currently would).